tests/test_service.py

```diff
@@ -473,6 +480,77 @@ def testGetAlertsAllInfo(self, mockGet):
                 self.assertTrue(isinstance(obj, Alert))
         self.assertIn((os.path.join(endpoint, "alerts/allinfo"),), tuple(mockGet.call_args))
 
+    # Test items() where get_all_path is the allinfo one
+    @mock.patch('requests.Session.get', return_value=MockResponse(json.dumps([alert_all_info_D, alert_all_info_2_D]), 200))
+    def testGetItemsAllInfo(self, mockGet):
+        self.assertEquals(len(mockGet.call_args_list), 0)
+        alertClient = self.argus.alerts
+        alertClient.set_get_all_path("alerts/allinfo")
+        alertClient.set_get_all_path_params(dict(shared=False))
```

Comment:
The usage here is not conducive for a shared client object. Also, a subsequent usage of `alerts` will continue to use the previous data that was set so it would have to explicitly clear them. Do you have a sample for the real usage to get a better idea of these methods are going to be used?

Reply:
Yes the usage is on line 492 - calling items() will call the 'get_all_path' endpoint, which here will be alerts/allinfo instead of the default "alerts" since I have set the path as alerts/allinfo. I kept "alerts" as the default path so that it is backwards compatible. 

Re " Also, a subsequent usage of alerts will continue to use the previous data that was set so it would have to explicitly clear them. " - I thought the point of the _coll is to be a cache so that you keep using the data in it and not reset it. This is the behavior from before my changes as well.

Re "The usage here is not conducive for a shared client object." - I think the alternative would be passing in the endpoint to each method that uses the 'get_all_path', which are items(), keys(), values(), and I think the endpoint would be the same across these methods, so to me it seems redundant, compared to setting it as a field in the client object.

Reply:
The service client objects that are accessed via `argus` object, such as `argus.alerts` are sort of "global" service objects and will be shared across all callers. It is not correct to add have setters and use them to modify these shared objects. What you should is to create new `AlertsServiceClient` objects as needed with modified behavior on a use and throw basis.

> I thought the point of the _coll is to be a cache so that you keep using the data in it and not reset it. This is the behavior from before my changes as well.

But the cache that would be generated is based on the modified fields and would be invalid for the default field values.

> I think the alternative would be passing in the endpoint to each method that uses the 'get_all_path'

Not really, if you follow my above suggestion. Support passing custom `get_all_path` and `get_all_path_params` via `AlertsServiceClient`'s constructor and use that to construct a new object where needed. E.g., the above would become something like, `alertClient = AlertsServiceClient(argus, get_all_path="alerts/allinfo", get_all_path_params=dict(shared=False))`.

Reply:
Ah, okay, I can see that. Thanks for the explanation! I've updated the code.